From 2a79e13d736ada7bb78b6bd0918747f753271c90 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:09 -0400
Subject: [tools] perf/trace: Beautify mlock & friends 'addr' arg

Message-id: <1408462094-14194-141-git-send-email-jolsa@redhat.com>
Patchwork-id: 88110
O-Subject: [PATCH RHEL7.1 BZ1131394 140/325] perf trace: Beautify mlock & friends 'addr' arg
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 3d903aa74a98da2836249bce86aaf557781239c9
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Sep 24 00:09:38 2013 -0300

description
===========
Printing it as an hex number.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 24b9370..2cf34fe 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -635,6 +635,10 @@ static struct syscall_fmt {
  { .name     = "madvise",    .errmsg = true,
    .arg_scnprintf = { [0] = SCA_HEX,  /* start */
         [2] = SCA_MADV_BHV, /* behavior */ }, },
+ { .name     = "mlock",     .errmsg = true,
+   .arg_scnprintf = { [0] = SCA_HEX, /* addr */ }, },
+ { .name     = "mlockall",   .errmsg = true,
+   .arg_scnprintf = { [0] = SCA_HEX, /* addr */ }, },
  { .name     = "mmap",     .hexret = true,
    .arg_scnprintf = { [0] = SCA_HEX,   /* addr */
         [2] = SCA_MMAP_PROT, /* prot */
@@ -645,6 +649,8 @@ static struct syscall_fmt {
  { .name     = "mremap",     .hexret = true,
    .arg_scnprintf = { [0] = SCA_HEX, /* addr */
         [4] = SCA_HEX, /* new_addr */ }, },
+ { .name     = "munlock",    .errmsg = true,
+   .arg_scnprintf = { [0] = SCA_HEX, /* addr */ }, },
  { .name     = "munmap",     .errmsg = true,
    .arg_scnprintf = { [0] = SCA_HEX, /* addr */ }, },
  { .name     = "open",     .errmsg = true,
-- 
1.7.1